### PR TITLE
Added support for anthropic and others

### PR DESCRIPTION
**Pull request by SamPink**
Using the LLM abstraction of llamaindex i've added support for more models. 

I tried to keep everything else the same as is for simplicity. 

I've already tested "anthropic:claude-3-haiku-20240307" vs groq:gemma-7b-it - haiku won!

Note: im not sure how  phospho.lab,  get_provider_and_model, get_sync_client  was being used

### CODE REVIEW

**Comment by oulianov** [comment]
Very cool ! Can you please remove the .vscode folder from your PR ?

**Comment by SamPink** [comment]
Done this now!

File: agent/llm.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def get_client(model_str):
+    provider, model = model_str.split(":")
```
Comment (oulianov) [inline agent/llm.py:2]: Here, you need to make sure model names with a ":" in it also works
Use https://github.com/phospho-app/phospho/blob/dev/phospho-python/phospho/lab/language_models.py#L29 as a reference

```
split_result = model.split(":")
if len(split_result) == 1:
    # Assume default provider to be openai
    provider = "openai"
    model_name = split_result[0]
elif len(split_result) > 2:
    # Some model names have :, so we need to join the rest of the string
    provider = split_result[0]
    model_name = ":".join(split_result[1:])
else:
    provider = split_result[0]
    model_name = split_result[1]
```

Comment (SamPink) [reply agent/llm.py]: Should be correct now

**Comment by SamPink** [comment]
Didn't do that properly, just fixed the bigs there!

**Comment by oulianov** [comment]
Cool ! Thank you again for your help.

**Pull request merged**